js/dataloader.js

```diff
@@ -11,17 +11,77 @@ var Utils = require('./utils.js');
  	Class: DataLoader
   		handles all loading of the data external
  		servers, transformations.
- 
+
  	Parameters:
  		serverUrl - sim server url
- 		simSearchQuery - sim search query specific url string
+ 		useSimSearchQuery - whether to perform compare (false) or search (true)
         limit - cutoff number
  */
- 
+
+function isBioLinkServer(serverURL) {
+	return serverURL.indexOf('https://api.monarchinitiative.org') === 0;
+}
+
+function buildSearchQuery(useBioLink, inputItemsString, qrySourceList) {
+	// console.log('buildSearchQuery', useBioLink, inputItemsString, qrySourceList);
+	if (useBioLink) {
+	    var result = inputItemsString + qrySourceList.join("&id=");
+	    // console.log('bsq', inputItemsString, qrySourceList, result);
+	}
+	else {
+	    var result = inputItemsString + qrySourceList.join("+");
+	}
+	return result;
+}
+
+function buildCompareQuery(useBioLink, sourceList, geneList) {
+	console.log('buildCompareQuery', useBioLink, sourceList, geneList);
+	if (useBioLink) {
+		var result = sourceList.join("+") + '/' + geneList.join(",");
+	}
+	else {
+		var result = sourceList.join("+") + '/' + geneList.join(",");
+	}
+	return result;
+}
+
 // Define the DataLoader constructor using an object constructor function
-var DataLoader = function(serverURL, simSearchQuery, limit) {
-	this.serverURL = serverURL;	
-    this.simSearchQuery = simSearchQuery; // object
+var DataLoader = function(serverURL, useSimSearchQuery, limit) {
+	this.serverURL = serverURL;
+	var useBioLink = isBioLinkServer(serverURL);
+
+	if (useSimSearchQuery) {
+		if (useBioLink) {
+			// https://api.monarchinitiative.org/api/sim/search?id=HP%3A0000006&id=HP%3A0000174&limit=100
+			this.simQuery = { // HTTP POST
+		        URL: '/api/sim/search',
+		        inputItemsString: 'id=', // HTTP POST, body parameter
+		        targetSpeciesString: '&taxon=', // HTTP POST, body parameter
+		        limitString: '&limit'
+		    };
+		}
+		else {
+			this.simQuery = { // HTTP POST
+		        URL: '/simsearch/phenotype',
+		        inputItemsString: 'input_items=', // HTTP POST, body parameter
+		        targetSpeciesString: '&target_species=', // HTTP POST, body parameter
+		        limitString: '&limit'
+		    };
+		}
+	}
+	else {
+		if (useBioLink) {
+			this.simQuery = { // compare API takes HTTP GET, so no body parameters
+                URL: '/compare' // used for owlSimFunction === 'compare' and genotype expansion compare simsearch - Joe
```

Comment:
Write better comment here to indicate that BioLink Compare has not been tested or utilized.